defender/staged/downloaded/Compound_Gasless_Voting.spec.js

```diff
@@ -0,0 +1,264 @@
+const mockContract = {
+  castVoteBySig: jest.fn(),
+  delegateBySig: jest.fn(),
+};
+
+// mock ethers
+jest.mock('ethers', () => ({
+  ...jest.requireActual('ethers'),
+  Contract: jest.fn().mockReturnValue(mockContract),
+}));
+
+const { ethers } = require('ethers');
+
+// mock the defender-relay-client package
+const mockProvider = {
+  getBalance: jest.fn().mockReturnValue(10),
+};
+const mockSigner = {
+  getAddress: jest.fn().mockReturnValue(ethers.constants.AddressZero),
+};
+
+jest.mock('defender-relay-client/lib/ethers', () => ({
+  DefenderRelayProvider: jest.fn().mockReturnValue(mockProvider),
+  DefenderRelaySigner: jest.fn().mockReturnValue(mockSigner),
+}));
+
+const { handler } = require('./Compound_Gasless_Voting');
+
+function createAutotaskEvent(batchedTxs) {
+  const autotaskEvent = {
+    request: {
+      body: batchedTxs,
+    },
+  };
+  return autotaskEvent;
+}
+
+function createCastVoteObject(address, proposalId, support, v, r, s) {
+  const voteObject = {
+    address,
+    proposalId,
+    support,
+    v,
+    r,
+    s,
+  };
+
+  return voteObject;
+}
+
+function createDelegateVoteObject(address, delegatee, nonce, expiry, v, r, s) {
+  const delegateVoteObject = {
+    address,
+    delegatee,
+    nonce,
+    expiry,
+    v,
+    r,
+    s,
+  };
+
+  return delegateVoteObject;
+}
+
+describe('check autotask', () => {
```

Comment:
The mocked functions from each test are persisting into the next test. For example, when the return value for `mockContract.state` is set to `1` in the first test case, it will still be `1` for the second test case if the mocked function is not reset.
```suggestion
describe('check autotask', () => {
  beforeEach(() => {
    mockContract.getCurrentVotes.mockReset();
    mockContract.state.mockReset();
  });
  
```

Reply:
added